components/utils/url.js

```diff
@@ -1,17 +1,11 @@
-export function getBaseURL() {
-  let protocol = null;
-  if (typeof window !== "undefined") {
-    protocol = window.location.protocol;
-  } else if (process.env.ENV_PROTOCOL) {
-    protocol = process.env.ENV_PROTOCOL;
-  } else {
-    protocol = process.env.NODE_ENV === "development" ? "http" : "https";
-  }
+export function getBaseUrl() {
+  const environment = process.env.NEXT_PUBLIC_ENVIRONMENT ?? "development";
 
-  const NEXT_PUBLIC_VERCEL_URL = process.env.NEXT_PUBLIC_VERCEL_URL;
-  const URL = `${protocol}://${NEXT_PUBLIC_VERCEL_URL}`;
-
-  return URL;
+  return {
+    production: `https://${process.env.NEXT_PUBLIC_DEPLOYMENT_URL}`,
```

Comment:
This change isn't really safe to make because it prevents me from testing production builds locally (or running the e2e test harness). Can you revert it?